algorithms/src/snark/varuna/ahp/matrices.rs

```diff
@@ -189,30 +173,54 @@ pub(crate) fn matrix_evals<F: PrimeField>(
 }
 
 // TODO for debugging: add test that checks result of arithmetize_matrix(M).
-pub(crate) fn arithmetize_matrix<F: PrimeField>(
-    id: &CircuitId,
-    label: &str,
-    matrix_evals: MatrixEvals<F>,
-) -> Result<MatrixArithmetization<F>> {
-    ensure!(matrix_evals.row_col.is_some(), "row_col evaluations are not available");
-
-    let interpolate_time = start_timer!(|| "Interpolating on K");
-    let row = matrix_evals.row.clone().interpolate();
-    let col = matrix_evals.col.clone().interpolate();
-    let row_col = matrix_evals.row_col.as_ref().unwrap().clone().interpolate();
-    let row_col_val = matrix_evals.row_col_val.clone().interpolate();
-    end_timer!(interpolate_time);
-
-    let label = &[label];
-    let mut labels = AHPForR1CS::<F, VarunaHidingMode>::index_polynomial_labels(label, std::iter::once(id));
-
-    Ok(MatrixArithmetization {
-        row: LabeledPolynomial::new(labels.next().unwrap(), row, None, None),
-        col: LabeledPolynomial::new(labels.next().unwrap(), col, None, None),
-        row_col: LabeledPolynomial::new(labels.next().unwrap(), row_col, None, None),
-        row_col_val: LabeledPolynomial::new(labels.next().unwrap(), row_col_val, None, None),
-        evals_on_K: matrix_evals,
-    })
+/// Contains information about the arithmetization of the matrices
+#[derive(Clone, Debug, CanonicalSerialize, CanonicalDeserialize, PartialEq, Eq)]
+pub struct MatrixArithmetization<F: PrimeField> {
+    /// LDE of the row indices of M^*.
+    pub row: LabeledPolynomial<F>,
+    /// LDE of the column indices of M^*.
+    pub col: LabeledPolynomial<F>,
+    /// LDE of the vector containing entry-wise products of `row` and `col`.
+    pub row_col: LabeledPolynomial<F>,
+    /// LDE of the vector containing entry-wise products of `row`, `col` and the non-zero entries of M.
+    pub row_col_val: LabeledPolynomial<F>,
+}
+
+impl<F: PrimeField> MatrixArithmetization<F> {
+    /// Create a new MatrixArithmetization
+    pub fn new(id: &CircuitId, label: &str, matrix_evals: &MatrixEvals<F>) -> Result<MatrixArithmetization<F>> {
+        let interpolate_time = start_timer!(|| "Interpolating on K");
+        let non_zero_domain = matrix_evals.row.domain();
```

Comment:
This is a confusing name; let's add a `matrix_evals.domain()` method.

Reply:
Done 68f294cea